lix/packages/sdk/src/combine/combine-changes.ts

```diff
@@ -0,0 +1,106 @@
+import type { LixPlugin } from "../plugin.js";
+import type { Change } from "../schema.js";
+import type { Lix } from "../types.js";
+
+/**
+ * Combines the changes of two lixes into the target lix.
+ */
+export async function combineChanges(args: { target: Lix; source: Lix }): Promise<void> {
+	const toBeAppliedChanges: Record<LixPlugin["key"], Array<Change>> = {};
+
+	// TODO ideally a right join via sqlite attaching works
+	//      instead of fetching all changes from source
+	const sourceChanges = await args.source.db
+		.selectFrom("change")
+		.selectAll()
+		.execute();
+
+	// 1. Get the changes that exists in source but not in target
+	for (const change of sourceChanges) {
+		const targetChange = await args.target.db
+			.selectFrom("change")
+			.select("id")
+			.where("id", "=", change.id)
+			.executeTakeFirst();
+
+		// change already exists, skip
+		if (targetChange !== undefined) {
+			continue;
+		}
+		// copy change to target
+		await args.target.db
+			.insertInto("change")
+			.values({
+				...change,
+				// @ts-expect-error - fix auto serialization
+				value: JSON.stringify(change.value),
+			})
+			.execute();
+
+		// 2. Check plugin if conflict exists
+		const plugin = args.source.plugins.find((p) => p.key === change.plugin_key);
+
+		if (!plugin) {
+			throw new Error("Plugin not found");
+		}
+
+		if (!plugin.reportConflicts) {
+			throw new Error("Plugin does not support conflict reporting");
+		}
+
+		const conflict = await plugin.reportConflicts({
+			sourceLix: args.source,
```

Comment:
isnt n argument missing here? this looks like it regenerates all conflicts on every change but would always lead to the same list of conflicts?